strider/traversal.py

```diff
@@ -0,0 +1,60 @@
+"""Traversal."""
+
+class NoAnswersError(Exception):
+    """No answers can be found."""
+
+
+def traverse_edge(qgraph: dict, qedge_id):
+    """Traverse an edge and return the resulting qgraph.
+
+    Remove the edge from the qgraph and pin both of its endpoints.
+    """
+    endpoints = (
+        qgraph["edges"][qedge_id]["subject"],
+        qgraph["edges"][qedge_id]["object"],
+    )
+    return {
+        "nodes": {
+            qnode_id: (qnode | {"ids": True} if qnode_id in endpoints else qnode)
+            for qnode_id, qnode in qgraph["nodes"].items()
+        },
+        "edges": {
+            _qedge_id: qedge
+            for _qedge_id, qedge in qgraph["edges"].items()
+            if _qedge_id != qedge_id
+        }
+    }
+
+
+def get_traversals(qgraph: dict):
```

Comment:
Traversals can only include edges connected to pinned nodes? I would include that in the documentation if that's the case.

Reply:
77c3964a535622072698e8be3c438ec4ab04dae6